suite/tests/CMakeLists.txt

```diff
@@ -4725,6 +4725,17 @@ if (BUILD_CLIENTS)
           "${drcachesim_path}@-simulator_type@record_filter@-filter_encodings2regdeps@-indir@${testname}.s*.dir/trace@-core_sharded@-cores@4@-outdir@${testname}.filtered.dir"
           # We run the opcode_mix analyzer to test econdings2regdeps filtered traces.
           "opcode_mix")
+
+      # Run the invariant_checker on a regdeps filtered trace.
```

Comment:
The `ci_shared_app` regdeps filtered trace doesn't fail on all invariant checks I've seen failing in our internal, more complex, "sample traced app", but it still fails on most of them (without the changes of this PR).
Here's a list:
#line on invariant_checker.cpp, CHECK, [OK | FAIL]
191, leftover expected_read_records_ expected_write_records_, OK
521, Maybe-blocking marker not preceded by syscall marker, FAIL
836, Missing read records, FAIL
845, Missing write records, FAIL
1195, decrement, OK
1215, decrement, OK
1530, branch doesn't go to correct target, FAIL

Reply:
Use the `kernel_xfer_app` which on UNIX is similar to the internal and has more types of events than the hello,world app.

Reply:
A simple swap from `ci_shared_app` to `kernel_xfer_app` results in:
```
  *** postcmd4 failed (1): ERROR: failed to initialize analyzer: Directory
  setup failed: Failed to open module file
  tool.invariant_checker_on_regdeps_trace.filtered.dir/raw/modules.log

  ***
```
I'll investigate what's happening.

Reply:
Solved.